core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/cluster/ClusterV4Request.java

```diff
@@ -30,17 +30,18 @@ public class ClusterV4Request implements JsonEntity {
     @ApiModelProperty(hidden = true)
     private String name;
 
-    @Size(max = 15, min = 5, message = "The length of the username has to be in range of 5 to 15")
+    @Size(max = 15, min = 5, message = "The length of the username has to be in the range of 5 to 15 characters")
     @Pattern(regexp = "(^[a-z][-a-z0-9]*[a-z0-9]$)",
-            message = "The username can only contain lowercase alphanumeric characters and hyphens and has start with an alphanumeric character")
+            message = "The username may only contain lowercase alphanumeric characters and hyphens, has to start with a letter and end with an " +
+                    "alphanumeric character")
     @ApiModelProperty(value = StackModelDescription.USERNAME)
     private String userName;
 
     @Pattern.List({
             @Pattern(regexp = "^.*[a-zA-Z].*$", message = "The password should contain at least one letter."),
             @Pattern(regexp = "^.*[0-9].*$", message = "The password should contain at least one number.")
     })
-    @Size(max = 100, min = 8, message = "The length of the password has to be in range of 8 to 100")
+    @Size(max = 100, min = 8, message = "The length of the password has to be in the range of 8 to 100 characters")
```

Comment:
not sure this is relevant to the ticket id which is in the commit message

Reply:
Well it was `ClusterV4Request.password` where the validation error was encountered. I wanted to fix the related error messages here and also in `DistroXClusterV1Request` as both seem to rely on the same restrictions. They definitely do not influence the password generation behavior, I agree. Shall I move these message changes to a separate PR then?

Reply:
Okay, the message fixes have been moved over to a separate commit under this PR. :smile: